plugins/Overlay/javascripts/Piwik_Overlay.js

```diff
@@ -196,9 +196,8 @@ var Piwik_Overlay = (function () {
             }
             $iframe.attr('src', iframeUrl);
             showLoading();
-        } else {
-            loadSidebar(location);
         }
+        loadSidebar(location);
```

Comment:
This should not be a bug to be fixed. This will always load the sidebar even if the page opened in the overlay is not configured properly, e.g. if there is no tracker.

It should already work "as expected" and not show the sidebar until the tracker in the iframe has messaged the top frame. Once loaded it will show all information in the sidebar instead of now suggesting it works while it does not.

Whether that message reaches the top frame is a separate issue that might need to be investigated.

Reply:
Yeah you are right, and thanks for checking so thoroughly. I've removed the 'fix' and will raise an issue soon about this problem in general.
So for now the scope of this change is back to, tests happy on 1st of month.